### PR TITLE
Work around for issues with basic http auth.

### LINKED ISSUE #3016: Multiple prompts to enter Basic HTTP Auth credentials
To reproduce:
Add a site that is secured with Basic HTTP Auth.  You will be prompted to enter credentials when you add the site to the app. 
Once added, tap to view the site's details screen (the list of posts, pages, comments, etc.) 
You will encounter multiple prompts to enter credentials, one for each request being made.  
View the post list and pull to refresh.  You are again prompted to enter credentials. 

It seems like every unique URL request eventually triggers the check `WPHTTPRequestOperation connection:willSendRequestForAuthenticationChallenge:` but subsequent requests of a previously requested URL do not.  I haven't gotten to the root of why subsequent requests do not show the prompt (maybe caching related?). 

Also, as far as I can tell we're not adding an authorization header to any request on the secured site and I'm not sure why we wouldn't be doing that. 

I used our site on bh.wpmt.co when testing.

### PR DESCRIPTION
**Pull request by aerych**
This is the same as #3031, but targeting the new branch.

Closes #3016

### CODE REVIEW

**Comment by aerych** [comment]
cc @sendhil

File: WordPress/Classes/Services/BlogService.m [hunk 1]
```diff
@@ -244,6 +249,60 @@ - (void)syncBlog:(Blog *)blog success:(void (^)())success failure:(void (^)(NSEr
...
+{
+    if (blog.account.isWpcom || blog.jetpackAccount) {
+        return NO;
+    }
+
+    __block BOOL batch = NO;
+    NSURL *url = [NSURL URLWithString:blog.url];
+    [[[NSURLCredentialStorage sharedCredentialStorage] allCredentials] enumerateKeysAndObjectsUsingBlock:^(NSURLProtectionSpace *ps, NSDictionary *dict, BOOL *stop) {
+        [dict enumerateKeysAndObjectsUsingBlock:^(id key, NSURLCredential *credential, BOOL *stop) {
+            DDLogVerbose(@"Removing credential %@ for %@", [credential user], [ps host]);
```
Comment (sendhil) [inline WordPress/Classes/Services/BlogService.m]: Removing credential?

Comment (aerych) [inline WordPress/Classes/Services/BlogService.m]: Bah! Forgot to remove that. Will update.

File: WordPress/Classes/Services/BlogService.m [hunk 2]
```diff
@@ -244,6 +249,60 @@ - (void)syncBlog:(Blog *)blog success:(void (^)())success failure:(void (^)(NSEr
...
+{
+    __weak __typeof(self) weakSelf = self;
+
+    PostService *postService = [[PostService alloc] initWithManagedObjectContext:self.managedObjectContext];
+
+    [postService syncPostsOfType:PostServiceTypePage forBlog:blog success:^{
+
+        [postService syncPostsOfType:PostServiceTypePost forBlog:blog success:^{
+
+            id<BlogServiceRemote> remote = [weakSelf remoteForBlog:blog];
```
Comment (sendhil) [inline WordPress/Classes/Services/BlogService.m]: Shouldn't the `weakSelf` go strong here? e.g. `__typeof (self) strongSelf = weakSelf;`

File: WordPress/Classes/Services/BlogService.m [hunk 3]
```diff
@@ -244,6 +249,60 @@ - (void)syncBlog:(Blog *)blog success:(void (^)())success failure:(void (^)(NSEr
     }
 }
 
+- (void)batchSyncBlog:(Blog *)blog
```
Comment (sendhil) [inline WordPress/Classes/Services/BlogService.m]: Not to nitpick here, but this doesn't seem so much like batching as it is staggering. Thoughts on renaming?

**Comment by sendhil** [comment]
Looks great.

**Pull request merged**